kernel/kernel-api/src/main/java/io/delta/kernel/internal/lang/FilteredCloseableIterator.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.internal.lang;
+
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import io.delta.kernel.utils.CloseableIterator;
+
+public abstract class FilteredCloseableIterator<RETURN_TYPE, ITER_TYPE>
```

Comment:
was this used anywhere?

Reply:
It is used by the `ReverseActionsToAddFilesIterator`.

Reply:
No filtering yet.